### ISSUE TITLE
can't get xbox on off to work.

### ISSUE THREAD

**Issue by Subivan1** [grzegorz914/homebridge-xbox-tv#207]
I beleive I have set up Xbox for remote access, and I can turn on and off from the "xbox app" but I cannot get the on  / off to work from HomeKit. I have followed the instructions using the aurthorisation manager and everything appears to have been done correctly but still no joy.

The log shows 

When the Xbox is off
[05/09/2024, 15:39:10] [homebridge-xbox-tv] Launched child bridge with PID 6157
[05/09/2024, 15:39:11] [homebridge-xbox-tv] Loaded homebridge-xbox-tv v3.1.2 child bridge successfully
[05/09/2024, 15:39:11] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, remote management not enabled, please check your console settings.
[05/09/2024, 15:39:11] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, {}
[05/09/2024, 15:39:21] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, published as external accessory.
[05/09/2024, 15:39:29] [homebridge-xbox-tv] warn: 192.168.4.83 X-Box, set Power, error: Error: send command type: Power, command: WakeUp, params: , error: AxiosError: Request failed with status code 404

showing remote management not enabled, yet as far as I can tell it is enabled.
![Xbox 1 remote access](https://github.com/user-attachments/assets/4bbe9fdc-e3f8-4d6e-9aae-1200a1ab94c9)
remote features are enabled.
![Xbox power](https://github.com/user-attachments/assets/90aa8c21-8522-4d40-9c91-d5abb768350b)
Power is set to sleep

When the xbox is on, i get slightly different logs.
[05/09/2024, 15:47:40] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, remote management not enabled, please check your console settings.
[05/09/2024, 15:47:40] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, {}
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, published as external accessory.
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Connected.
[05/09/2024, 15:47:50] [homebridge-xbox-tv] -------- X-Box --------'
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Manufacturer: Microsoft
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Model: Xbox
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Serialnr: MY USER IS
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Firmware: 10.0.26100
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Locale: en-GB
[05/09/2024, 15:47:50] [homebridge-xbox-tv] ----------------------------------
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Power: ON
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Input Name: Settings
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Reference: Microsoft.Xbox.Settings_8wekyb3d8bbwe!Xbox.Settings.Application
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Title Id: 3317923840
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Product Id: Settings
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Volume: 0%
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Mute: ON
[05/09/2024, 15:47:50] [homebridge-xbox-tv] Device: 192.168.4.83 X-Box, Media State: STOPPED

Once again it states remote management not enabled yet it also shows connected and the button in homekit shows the correct state of the xbox as on.
trying to turn off the box from homekit results in 
[05/09/2024, 15:50:06] [homebridge-xbox-tv] warn: 192.168.4.83 X-Box, set Power, error: Error: send command type: Power, command: TurnOff, params: , error: AxiosError: Request failed with status code 404

So far i have looked at other issues and attempted solutions that are discussed, including resetting the xbox, trying different versions of Xbox-tv, resetting the authorisiation manager and clearing the web api token but no joy. Check state button shows Console authorized and activated but still no joy in making the on off button work.

I am probably missing somehting really simple and straight forward but i am scratching my head here. Any suggestions on what i am doing wrong?

**Comment by grzegorz914** [comment]
Are you sure you have activated correct Xbox? Are you set correct xboxLiveId, can you post whole log after save config and restart plugin, please enable debug mode in plugin config advanced settings>>>system>>>debug

**Comment by Subivan1** [comment]
I only have one xbox and i have checked that the correct live i is used.
[homebridge-xbox-tv latest.log.txt](https://github.com/user-attachments/files/16900556/homebridge-xbox-tv.latest.log.txt)

**Comment by grzegorz914** [comment]
```
[05/09/2024, 22:49:10] [homebridge-xbox-tv] Device: 192.168.4.83 Xbox, debug: getConsoleStatusData, result: {
  "status": {
    "errorCode": "XboxDataNotFound",
    "errorMessage": "Could not find xbox document"
  },
  "id": null,
  "name": null,
  "locale": null,
  "region": null,
  "consoleType": null,
  "powerState": null,
  "playbackState": null,
  "loginState": null,
  "focusAppAumid": null,
  "isTvConfigured": false,
  "digitalAssistantRemoteControlEnabled": null,
  "consoleStreamingEnabled": null,
  "remoteManagementEnabled": null
}
```
Your console is not register in xbox live, you need to check xboxLiveId and account where is registered.

**Comment by Subivan1** [comment]
I am not sure what you mean here. I am logged into xbox live (I have game pass) on the xbox,  my microsoft account allows me to remove or unlink from the account, which implies it is already linked. Where do i go to check it is registered ?

[All Devices](https://account.microsoft.com/devices)

XBOX-X

Remove this device
Unlink from your account

**Comment by Subivan1** [comment]
Solved it.
Thanks for the help, you were right about the xboxLiveId field.
I made an assumption that the live id requested in the authorisation process was actually my microsoft live id. When i reread the full description in the configuration section i noticed that it should have come from the console.

putting that in has resolved all the issues. 

Thanks for the help and sorry for making an assumption and causing you more work.
